base/compiler/abstractinterpretation.jl

```diff
@@ -542,7 +545,6 @@ function abstract_call_method(interp::AbstractInterpreter, method::Method, @nosp
                         parent_method2 isa Method || (parent_method2 = nothing) # Union{Method, Nothing}
                         if parent.linfo.def === sv.linfo.def && sv_method2 === parent_method2
                             topmost = infstate
-                            edgecycle = true
```

Comment:
What's the reason to delete these `edgecycle = true` ? 
In these cases there is a cycle, and we still need to try recursion detection in `abstract_call_method_with_const_args` ?

Reply:
Yes, you're right - these were deleted by accident (leftover from an earlier experiment) - thanks.